README.md

```diff
@@ -143,25 +143,20 @@ when a pull request contains unformatted code. Using `--write-mode=diff` instruc
 rustfmt to exit with an error code if the input is not formatted correctly.
 It will also print any found differences.
 
-(These instructions use the nightly version of Rustfmt. If you want to use the
-Syntex version replace `install rustfmt-nightly` with `install rustfmt`).
-
 A minimal Travis setup could look like this:
 
 ```yaml
 language: rust
-cache: cargo
 before_script:
-- export PATH="$PATH:$HOME/.cargo/bin"
-- which rustfmt || cargo install rustfmt-nightly
+- rustup toolchain install nightly
+- rustup run nightly rustup component add rustfmt-preview
```

Comment:
Could this line be just `rustup component add rustfmt-preview`?

Reply:
I think it could not, since the default toolchain may not be nightly.

Reply:
rustfmt-preview is available on every channel now, so I don't think that matters, I'm not 100% sure about the rustup logic here though

Reply:
> rustfmt-preview is available on every channel now

That is nice!

I am not certain how rustup works inside travis build, but if we simply write `rustup component add rustfmt-preview`, won't travis use different version of rustfmt for different channel?

Reply:
`rustup component add rustfmt-preview` will add rustfmt-preview to the default toolchain (or current toolchain if it is overridden for the current directory). To add it to the nightly toolchain, we should use `rustup component add --toolchain nightly rustfmt-preview`, and we should do that for each toolchain used by Travis (I don't think there is a way to install a component for all toolchains).

Reply:
I do not think we should add `rustfmt-preview` for each toolchain - we should only add a single version of rustfmt and use it in every toolchain. Otherwise, the result of `cargo fmt` will differ in each toolchain.